plugins/modules/gitlab_milestone.py

```diff
@@ -0,0 +1,499 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Gabriele Pongelli (gabriele.pongelli@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_milestone
+short_description: Creates/updates/deletes GitLab Milestones belonging to project or group.
+description:
+  - When a milestone does not exist, it will be created.
+  - When a milestone does exist, its value will be updated when the values are different.
+  - Milestones can be purged
+  - Task can be run in check mode
+author:
+  - "Gabriele Pongelli (@gpongelli)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  state:
+    description:
+      - Create or delete milestone.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  purge:
+    description:
+      - When set to V(true), delete all milestone which are not mentioned in the task.
+    default: false
+    type: bool
+    required: false
+  project:
+    description:
+      - The path and name of the project, mandatory field but it's alternatively with group.
+    required: false
+    type: str
+  group:
+    description:
+      - The path of the group, mandatory field but it's alternatively with project.
+    required: false
+    type: str
+  milestones:
+    version_added: 1.0.0
+    description:
+      - A list of dictionaries that represents gitlab project's or group's milestones.
+    type: list
+    elements: dict
+    required: false
+    default: []
+    suboptions:
+      title:
+        description:
+          - The name of the milestone.
+        type: str
+        required: true
+      due_date:
+        description:
+          - Milestone due date in YYYY-MM-DD format.
+        type: str
+        required: false
+        default: null
+      start_date:
+        description:
+          - Milestone start date in YYYY-MM-DD format.
+        type: str
+        required: false
+        default: null
+      description:
+        description:
+          - Milestone's description
+        type: str
+        default: null
+'''
+
+
+EXAMPLES = '''
+# same project's task can be executed for group
+- name: Create one milestone
+  community.general.gitlab_milestone:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: "group1/project1"
+    milestones:
+      - title: milestone_one
+        start_date: 2024-01-04
```

Comment:
This specifies a YAML date. Did you check that your code receives this as expected?

Reply:
No, it’s expected to be a string with YYYY-MM-DD format. I’ll add double quotes there, thanks

Reply:
Changed